test/Dialect/SystemC/module-error.mlir

```diff
@@ -154,3 +154,92 @@ systemc.module @invalidSignalOpReturnType () {
   // expected-error @+1 {{invalid kind of type specified}}
   %signal0 = systemc.signal : i32
 }
+
+// -----
+
+systemc.module @m1() {}
+
+// expected-note @+1 {{in module '@instanceDeclNameConflict'}}
+systemc.module @instanceDeclNameConflict () {
+  // expected-note @+1 {{'name' first defined here}}
+  %0 = "systemc.signal"() {name="name"} : () -> !systemc.signal<i32>
+  // expected-error @+1 {{redefines name 'name'}}
```

Comment:
It isn't strictly necessary that this errors. This could be unique during emission. What does SV do here?

Reply:
In which case would it be unique during emission?

Reply:
I think the emitter will just determine a unique name from this base name. This is in conflict with the emission infra you've been working on as it would require whole module information.

I am fine with keeping this as is.